### PR TITLE
win32: fix resizing window when maximized

### PR DESCRIPTION
**Pull request by kasper93**
This remembers and sets window-scale when exiting maximized state. Also fixes not staying in maximized state when exiting fullscreen mode.

Fixes: #14126

### CODE REVIEW

**Comment by kasper93** [comment]
@na-na-hi: I actually had this patch in my local branch for a long time, you could check if you'd like, but for me it was working fine for long time.

File: video/out/w32_common.c [hunk 1]
```diff
@@ -1449,7 +1449,6 @@ static LRESULT CALLBACK WndProc(HWND hWnd, UINT message, WPARAM wParam,
             w32->windowrc.right = w32->windowrc.left + w;
             w32->windowrc.bottom = w32->windowrc.top + h;
             signal_events(w32, VO_EVENT_RESIZE);
-            MP_VERBOSE(w32, "resize window: %d:%d\n", w, h);
```
[the remark this answers is not in the document]
Comment (kasper93) [reply video/out/w32_common.c:1452]: Because it spams and is not useful.

[the remark this answers is not in the document]
Comment (kasper93) [reply video/out/w32_common.c:1452]: done

**Comment by kasper93** [comment]
Maybe this time, it restores the size.

**Comment by kasper93** [comment]
> It still doesn't restore to the specified geometry for me.

@na-na-hi: I've taken a look again at this. It seems to work correctly for me.

1. set fullscreen yes
2. set window-maximized yes
3. set geometry 500x500
4. set fullscreen no
5. set window-maximized no

Window is 500x500 after this sequence. Could you help me reproduce the issue?

**Comment by kasper93** [comment]
@na-na-hi: Can we agree current state of this commit is an improvement over master? I'd like to merge it, if you don't mind. I'm struggling to make it work better and the more I test the more corner cases I found. We mix internal and external window state and I think more in-depth refactor is needed here to fix things fully.

**Comment by kasper93** [comment]
@na-na-hi forgot about this, could you test latest version?

**Comment by kasper93** [comment]
> With the newest version, the problem I mentioned before doesn't happen anymore, but now ` set window-maximized yes - set fullscreen yes - set fullscreen no` would unmaximize the window when exiting fullscreen, when it should return to maximized state.

Does it work now for you?

**Comment by verygoodlee** [comment]
brings a new issue, in `--no-title-bar` mode, unmaximize cannot restore the window to manually adjusted postion and size.
- `mpv any.jpg --no-title-bar --autofit-larger=70%x70% --geometry=50%:50% --pause`
- adjust window positon and size manually
- set window-maximized yes
- set window-maximized no

https://github.com/user-attachments/assets/ba436c06-a828-45da-9507-68bbd596d5b1
